src/core/time_arrays.jl

```diff
@@ -213,30 +353,77 @@ function Base.setindex!(mat::TimestepMatrix{VariableTimestep{D_TIMES}, T, 2}, va
+	LAST = FIRST + ((size(mat.data, 1) - 1) * STEP)
+	t = _get_time_value_position([FIRST:STEP:LAST...], ts)
+	setindex!(mat.data, val, idx, t)
+end
+
+function Base.setindex!(mat::TimestepMatrix{VariableTimestep{TIMES}, T_data, 2}, val, idx::AnyIndex, ts::TimestepValue{T_time}) where {T_data, TIMES, T_time}
+	t = _get_time_value_position(TIMES, ts)
+	setindex!(mat.data, val, idx, t)
+end
+
+function Base.setindex!(mat::TimestepMatrix, val, idx::AnyIndex, ts::TimestepIndex)
+	setindex!(mat.data, val, idx, ts.index)
+end
+
+function Base.setindex!(mat::TimestepMatrix, val, ts::TimestepIndex, idx::AnyIndex)
+	setindex!(mat.data, val, ts.index, idx)
+end
+
+# Colon support
+
+function Base.getindex(mat::TimestepMatrix{TS, T, ti}, idx1::Colon, idx2::Colon) where {TS<:AbstractTimestep, T, ti}
+	return mat.data[idx1, idx2]
+end
+
+function Base.setindex!(mat::TimestepMatrix{TS, T, ti}, val, idx1::Colon, idx2::Colon) where {TS<:AbstractTimestep, T, ti}
+	mat.data[idx1, idx2] .= val
+end
+
+
 # int indexing version supports old-style components and internal functions, not
 # part of the public API
 
-function Base.getindex(mat::TimestepMatrix{FixedTimestep{FIRST, STEP}, T, ti}, idx1::AnyIndex, idx2::AnyIndex) where {T, FIRST, STEP, ti}
+function Base.getindex(mat::TimestepMatrix{FixedTimestep{FIRST, STEP}, T, ti}, idx1::AnyIndex_NonColon, idx2::AnyIndex_NonColon) where {T, FIRST, STEP, ti}
+	_throw_int_getindex_depwarning()
 	return mat.data[idx1, idx2]
 end
 
-function Base.getindex(mat::TimestepMatrix{VariableTimestep{TIMES}, T, ti}, idx1::AnyIndex, idx2::AnyIndex) where {T, TIMES, ti}
+function Base.getindex(mat::TimestepMatrix{VariableTimestep{TIMES}, T, ti}, idx1::AnyIndex_NonColon, idx2::AnyIndex_NonColon) where {T, TIMES, ti}
+	_throw_int_getindex_depwarning()
 	return mat.data[idx1, idx2]
 end
 
 function Base.setindex!(mat::TimestepMatrix{FixedTimestep{FIRST, STEP}, T, ti}, val, idx1::Int, idx2::Int) where {T, FIRST, STEP, ti}
+	_throw_int_setindex_depwarning()
 	setindex!(mat.data, val, idx1, idx2)
 end
 
-function Base.setindex!(mat::TimestepMatrix{FixedTimestep{FIRST, STEP}, T, ti}, val, idx1::AnyIndex, idx2::AnyIndex) where {T, FIRST, STEP, ti}
+function Base.setindex!(mat::TimestepMatrix{FixedTimestep{FIRST, STEP}, T, ti}, val, idx1::AnyIndex_NonColon, idx2::AnyIndex_NonColon) where {T, FIRST, STEP, ti}
+	_throw_int_setindex_depwarning()
 	mat.data[idx1, idx2] .= val
 end
 
 function Base.setindex!(mat::TimestepMatrix{VariableTimestep{TIMES}, T, ti}, val, idx1::Int, idx2::Int) where {T, TIMES, ti}
+	_throw_int_setindex_depwarning()
 	setindex!(mat.data, val, idx1, idx2)
 end
 
-function Base.setindex!(mat::TimestepMatrix{VariableTimestep{TIMES}, T, ti}, val, idx1::AnyIndex, idx2::AnyIndex) where {T, TIMES, ti}
+function Base.setindex!(mat::TimestepMatrix{VariableTimestep{TIMES}, T, ti}, val, idx1::AnyIndex_NonColon, idx2::AnyIndex_NonColon) where {T, TIMES, ti}
+	_throw_int_setindex_depwarning()
 	mat.data[idx1, idx2] .= val
 end
```

Comment:
I didn't really think these through since they are deprecated, we haven't really been keeping them updated since we plan to delete them, but I could do that!

Reply:
Making a new Issue and PR to remove duplicate functions